master/pkg/schemas/expconf/hyperparameters_config.go

```diff
@@ -37,14 +63,62 @@ func (h *HyperparameterV0) UnmarshalJSON(data []byte) error {
 		return err
 	}
 	if _, ok := parsed.(map[string]interface{}); ok {
+		parsedMap := parsed.(map[string]interface{})
+		hpType, hasType := parsedMap["type"]
+		if (hasType && hpType.(string) == "object") || !hasType {
+			nestedHPs := make(map[string]HyperparameterV0)
+			if hasType {
+				delete(parsedMap, "type")
+			}
+			for key := range parsedMap {
+				bytes, _ := json.Marshal(parsedMap[key])
+				var hp HyperparameterV0
+				err := json.Unmarshal(bytes, &hp)
+				if err != nil {
+					return err
+				}
+				nestedHPs[key] = hp
+			}
+			h.RawNestedHyperparameter = &nestedHPs
+			return nil
+		}
 		return union.Unmarshal(data, h)
 	}
 	h.RawConstHyperparameter = &ConstHyperparameterV0{RawVal: parsed}
 	return nil
 }
 
+func hpsToMap(hps HyperparameterV0) (map[string]interface{}, error) {
+	output := make(map[string]interface{})
+	var err error
+	for key, hp := range *hps.RawNestedHyperparameter {
+		if hp.RawNestedHyperparameter != nil {
+			output[key], err = hpsToMap(hp)
+			if err != nil {
+				return nil, err
+			}
+		} else {
+			hpBytes, _ := hp.MarshalJSON()
+			var unionHP map[string]interface{}
+			err = json.Unmarshal(hpBytes, &unionHP)
+			if err != nil {
+				return nil, err
+			}
+			output[key] = unionHP
+		}
+	}
+	return output, nil
+}
+
 // MarshalJSON implements the json.Marshaler interface.
 func (h HyperparameterV0) MarshalJSON() ([]byte, error) {
+	if h.RawNestedHyperparameter != nil {
+		hps, err := hpsToMap(h)
+		if err != nil {
+			return []byte("null"), err
```

Comment:
I think it is preferable to return `nil, err` rather than `<valid json>, err`, for readability reasons.  I had a wtf moment where I thought you were dropping the error silently and marshaling everything as `"null"`, before looking closer.